ibis/expr/types/joins.py

```diff
@@ -205,8 +222,83 @@ def join(
         # return with a new JoinExpr wrapping the new join chain
         return self.__class__(left, collisions=collisions)
 
-    def select(self, *args, **kwargs):
-        """Select expressions."""
+    @functools.wraps(Table.asof_join)
+    def asof_join(  # noqa: D102
+        self: Table,
+        right: Table,
+        on,
+        predicates=(),
+        by=(),
```

Comment:
Is `by` used anywhere?

Reply:
Ah, nevermind I see it.

Reply:
Can we get rid of `by`? Having that alongside `predicates` and `on` seems a bit confusing.

Reply:
Created a follow up: https://github.com/ibis-project/ibis/issues/7869